force-app/main/default/classes/PS_GatewayManagement.cls

```diff
@@ -43,23 +43,58 @@ public with sharing class PS_GatewayManagement {
         };
         PS_IntegrationService.setConfiguration(gatewayConfig);
     }
-
     @AuraEnabled
+    public static void setGatewayAssignmentEnabled(Boolean gatewayAssignmentEnabled) {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        giftEntrySettings.Enable_Gateway_Assignment__c = gatewayAssignmentEnabled;
+
+        upsert giftEntrySettings;       
+    }
+
+    public static Boolean isGatewayAssignmentEnabled() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+
+        return giftEntrySettings.Enable_Gateway_Assignment__c;
+    }
+
     public static String getGatewayIdFromConfig() {
         PS_IntegrationServiceConfig.Service configService = new PS_IntegrationServiceConfig.Service();
 
         return configService.getGatewayIds();
     }
 
-    @AuraEnabled
```

Comment:
Can we make this method `private` since it is now only called in this class

Reply:
updated